tests/test_ble_lifecycle_controller.py

```
@@ -0,0 +1,689 @@
+
+
+class TestBLELifecycleControllerClientManagement:
+    """Test cases for client management methods."""
+
+    def test_disconnect_and_close_client_delegates_to_disconnect_coordinator(
+        self,
+    ) -> None:
+        """_disconnect_and_close_client should delegate to _disconnect coordinator."""
+        calls: list[object] = []
+        mock_disconnect = SimpleNamespace(
+            disconnect_and_close_client=lambda client: calls.append(client)
```

Ruff (0.15.6)

[warning] Lambda may be unnecessary; consider inlining inner function

Inline function call

(PLW0108)